File: benchmarks/c/format_hex.c
#include <stdint.h>
#include <stdio.h>
#include <string.h>

static int64_t format_hex_bench(int64_t iterations) {
    int64_t sink = 0;
    char buf[32];
    for (int64_t i = 0; i < iterations; i++) {
        snprintf(buf, sizeof(buf), "0x%llX", (unsigned long long)i);
        size_t n = strlen(buf);
        sink += (int64_t)n + (unsigned char)buf[n - 1];
    }
    return sink;
}

int main(void) {
    const int64_t iterations = 400000;
    int64_t result = format_hex_bench(iterations);
    printf("%lld\n", (long long)result);
    return 0;
}

File: benchmarks/python/format_hex.py
def format_hex_bench(iterations: int) -> int:
    sink = 0
    i = 0
    while i < iterations:
        s = hex(i).upper().replace("X", "x")
        sink += len(s) + ord(s[-1])
        i += 1
    return sink


def main() -> int:
    iterations = 400000
    result = format_hex_bench(iterations)
    print(result)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
